metrics/core/lib/src/util/validation/printer/validation_result_printer.dart

```diff
@@ -0,0 +1,126 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:metrics_core/src/util/validation/target_validation_result.dart';
+import 'package:metrics_core/src/util/validation/validation_conclusion.dart';
+import 'package:metrics_core/src/util/validation/validation_result.dart';
+import 'package:metrics_core/src/util/validation/validation_target.dart';
+
+/// A class that is responsible for printing the [ValidationResult].
+class ValidationResultPrinter {
+  /// An instance of the [StringSink] this printer uses to print the result.
+  final StringSink sink;
+
+  /// Creates a new instance of the [ValidationResultPrinter] with the given
+  /// [StringSink].
+  ///
+  /// Throws an [ArgumentError] if the given [sink] is `null`;
+  ValidationResultPrinter({
+    this.sink,
+  }) {
+    ArgumentError.checkNotNull(sink, 'sink');
+  }
+
+  /// Prints the given [result] to the [sink].
+  void print(ValidationResult result) {
+    final validationResults = result.results;
+
+    final validationResultEntries = validationResults.entries;
+
+    for (final entry in validationResultEntries) {
+      final target = entry.key;
+      final result = entry.value;
+
+      final message = _buildTargetValidationOutput(target, result);
+
+      sink.writeln(message);
+    }
+  }
+
+  /// Builds the output that provides the information about
+  /// the given [target] and [validationResult].
+  String _buildTargetValidationOutput(
+    ValidationTarget target,
+    TargetValidationResult validationResult,
+  ) {
+    final buffer = StringBuffer();
+
+    _addConclusion(buffer, validationResult.conclusion);
+    buffer.write(' ${target.name}');
+    _addDescription(buffer, target.description);
+    _addResultDescription(buffer, validationResult.description);
+    _addDetails(buffer, validationResult.details);
+    buffer.write('\n');
+    _addContext(buffer, validationResult.context);
+
+    return buffer.toString();
+  }
+
+  /// Adds the [String] representation of the given [conclusion] to the
+  /// given [buffer].
+  ///
+  /// Records `[?]` if the given [ValidationConclusion.indicator]
+  /// is empty or `null`.
```

Comment:
🏷️ 
Please apply where necessary.
```suggestion
  /// Records `[?]` if the given [ValidationConclusion.indicator]
  /// is `null` or empty.
```